sklearn/linear_model/_sgd_fast.pyx

```diff
@@ -362,6 +362,34 @@ cdef class SquaredEpsilonInsensitive(Regression):
         return SquaredEpsilonInsensitive, (self.epsilon,)
 
 
+cdef class PinBall(Regression):
+    """Pin ball loss used for quantile regression.
+
+    loss = q * max(y - p, 0) + (1 - q) * max(p - y, 0)
+    """
+
+    cdef double quantile
+
+    def __init__(self, double quantile):
+        self.quantile = quantile
+
+    cdef double loss(self, double p, double y) nogil:
+        cdef double ret = y - p
+        return self.quantile * ret if ret > 0\
+         else (1 - self.quantile) * -ret
+
+    cdef double dloss(self, double p, double y) nogil:
+        if y - p > 0:
+            return -self.quantile
+        elif y - p < 0:
+            return 1- self.quantile
+        else:
+            return 0
```

Comment:
```suggestion
        if y >= p:
            return -self.quantile
        else:
            return 1.0 - self.quantile
```

Reply:
(Technically, both are correct as both are subgradients at y=p.)

Reply:
I would prefer to have gradient as 0 when `y==p`.

Reply:
Expecting `==` with floating numbers is questionable regarding numerical stability.
I would prefer to use this definition because it is also the definition of the loss in `sklearn/_loss/loss.py` so we will not have any change of behaviour once we will use the common losses.

Reply:
makes sense to me now.